src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -6837,25 +6837,35 @@ def merge_asof(
             right_frame.ordered_dataframe.to_projected_snowpark_dataframe()
         )
         if on:
-            on_quoted_identifier = quote_name_without_upper_casing(on)
-            lhs_expr = left_snowpark_df[on_quoted_identifier]
-            rhs_expr = right_snowpark_df[on_quoted_identifier]
-            order_by_expr = [
-                left_snowpark_df[on_quoted_identifier],
-                right_snowpark_df[on_quoted_identifier],
-            ]
+            left_on_quoted_identifier = (
```

Comment:
What happens if multiple columns share the same label in vanilla pandas? I think grabbing the first column is good for us, but I'd be curious to see how pandas handles that.

Reply:
Actually I made sure that we handle that case same as pandas, I can add a test for it

Reply:
native pandas returns `ValueError: The column label 'A' is not unique.`

Reply:
Currently in the code, we handle the duplicates though. 

```
    A left_val   A  right_val   A
0   1        a  10          1  10
1   5        b  12          3  13
2  10        c  13          7  15
```

I think this behavior is okay without any additional warnings, but I could also add checks for duplicates. What do you think?